File: LeetCode/C++/1. Two Sum.cpp
/// LeetCode 1. Two Sum by C++
/// write by LazyWolfLin in 16.08.29

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> indices;
        for (size_t i = 0; i < nums.size(); ++i)
        {
            auto iter = indices.find(target - nums[i]);
            if (iter != indices.end())
                return {iter->second, i};
            indices[nums[i]] = i;
        }
        return {};
    }
};

File: LeetCode/Python3/1. Two Sum.py
# LeetCode 1. Two Sum by Python3
# write by LazyWolfLin in 18.07.21

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        indices = dict()
        for i in range(0, len(nums)):
            if (target - nums[i]) in indices:
                return [indices[target - nums[i]], i]
            else:
                indices[nums[i]] = i;
        return [];